tests/run-make/non-unicode-in-incremental-dir/rmake.rs

```diff
@@ -0,0 +1,24 @@
+extern crate run_make_support;
+
+use run_make_support::{libc, rustc, tmp_dir};
+
+fn main() {
+    #[cfg(unix)]
+    let non_unicode: &std::ffi::OsStr = std::os::unix::ffi::OsStrExt::from_bytes(&[0xFF]);
+    #[cfg(windows)]
+    let non_unicode: std::ffi::OsString = std::os::windows::ffi::OsStringExt::from_wide(&[0xD800]);
+    match std::fs::create_dir(tmp_dir().join(&non_unicode)) {
+        Err(e) if e.raw_os_error() == Some(libc::EILSEQ) => {
```

Comment:
That error code assumes create_dir is implemented using a POSIX API. That isn't true on Windows and may not even be true on Unix OSes (depending on how the create_dir is implemented).

Reply:
I don't see `EILSEQ` referenced anywhere in `std` so I don't think there's a portable way to detect this. You can always try to create a file with a unicode path first to ensure the test doesn't silently fail for other permissions reasons.